### PR TITLE
linuxKernel.packages.linux_6_10.nvidia_x11_legacy390: compatible with kernel <= 6.10

### PR DESCRIPTION
**Pull request by drupol**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by drupol** [comment]
@Kiskae Do you have a clue on how to fix this?

**Comment by Kiskae** [comment]
> @Kiskae Do you have a clue on how to fix this?

If you look in that repository you need to apply the patches in order. You only applied the patch for linux 6.10, so all the breakages between 6.1 and 6.10 are still broken.

Looking at `PKGBUILD` this is the order:

>     # Restore phys_to_dma support (still needed for 390.138)
>     # From loqs via https://bugs.archlinux.org/task/58074
>     patch -Np1 -i ../kernel-4.16+-memory-encryption.patch
> 
>     # From Joan Bruguera via Ike Devolder
>     patch -Np1 -i ../kernel-6.2.patch
> 
>     # From Ike Devolder
>     patch -Np1 -i ../kernel-6.3.patch
> 
>     # From Joan Bruguera via duht
>     patch -Np1 -i ../kernel-6.4.patch
> 
>     # From Joan Bruguera via Witscher
>     patch -Np1 -i ../kernel-6.5.patch
> 
>     # From canolucas
>     patch -Np1 -i ../kernel-6.6.patch
> 
>     # From Joan Bruguera
>     patch -Np1 -i ../kernel-6.8.patch
> 
>     # From loqs
>     patch -Np1 -i ../gcc-14.patch
> 
>     # From Ike Devolder, edited by drankinatty
>     patch -Np1 -i ../kernel-6.10.patch

**Comment by drupol** [comment]
@Kiskae Good for you now? For me it's fine as it is now.

**Comment by Kiskae** [comment]
> @Kiskae Good for you now? For me it's fine as it is now.

Post the result of `NIXPKGS_ALLOW_UNFREE=1 nix run nixpkgs#nixpkgs-review -- pr --checkout commit 337062 --extra-nixpkgs-config '{ nvidia.acceptLicense = true; }' --eval local` so we can verify some obscure kernel isn't being broken, and then it looks good to me.

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337062` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 337062` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)

**Comment by drupol** [comment]
Passed:

```
❯ NIXPKGS_ALLOW_UNFREE=1 nix run nixpkgs#nixpkgs-review -- pr --checkout commit 337062 --extra-nixpkgs-config '{ nvidia.acceptLicense = true; }' --eval local
$ git -c fetch.prune=false fetch --no-tags --force https://github.com/NixOS/nixpkgs master:refs/nixpkgs-review/0 pull/337062/head:refs/nixpkgs-review/1
remote: Enumerating objects: 1526, done.
remote: Counting objects: 100% (1340/1340), done.
remote: Compressing objects: 100% (532/532), done.
remote: Total 1526 (delta 1032), reused 990 (delta 783), pack-reused 186 (from 1)
Receiving objects: 100% (1526/1526), 2.00 MiB | 7.56 MiB/s, done.
Resolving deltas: 100% (1057/1057), completed with 217 local objects.
From https://github.com/NixOS/nixpkgs
   54bb59c4041b..2b97f5f5bb51  master                -> refs/nixpkgs-review/0
 + 6d34f8cdc87f...55450445592d refs/pull/337062/head -> refs/nixpkgs-review/1  (forced update)
$ git worktree add /home/pol/.cache/nixpkgs-review/pr-337062/nixpkgs eb0e6df0cdd2641ec0651cd9802ff4cff3b3e915
Preparing worktree (detached HEAD eb0e6df0cdd2)
Updating files: 100% (42902/42902), done.
HEAD is now at eb0e6df0cdd2 Merge pull request #336860 from r-ryantm/auto-update/resources
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/pol/.cache/nixpkgs-review/pr-337062/nixpkgs nixpkgs-overlays=/tmp/tmpryw4j5el -qaP --xml --out-path --show-trace --no-allow-import-from-derivation
$ git merge --no-commit --no-ff 55450445592d7369720c94fbe18c7f809b82ce8f
Automatic merge went well; stopped before committing as requested
$ nix-env --extra-experimental-features no-url-literals --option system x86_64-linux -f <nixpkgs> --nix-path nixpkgs=/home/pol/.cache/nixpkgs-review/pr-337062/nixpkgs nixpkgs-overlays=/tmp/tmpryw4j5el -qaP --xml --out-path --show-trace --no-allow-import-from-derivation --meta
21 packages updated:
nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11 nvidia-x11

$ nix build --file /nix/store/s5dnqifxhlqrd8zicabxswwdgjlhplvq-nixpkgs-review-2.10.5/lib/python3.12/site-packages/nixpkgs_review/nix/review-shell.nix --nix-path 'nixpkgs=/home/pol/.cache/nixpkgs-review/pr-337062/nixpkgs nixpkgs-overlays=/tmp/tmpryw4j5el' --extra-experimental-features 'nix-command no-url-literals' --no-link --keep-going --no-allow-import-from-derivation --option build-use-sandbox relaxed --argstr system x86_64-linux --argstr nixpkgs-path /home/pol/.cache/nixpkgs-review/pr-337062/nixpkgs --argstr nixpkgs-config-path /tmp/tmp129cio38.nix --argstr attrs-path /home/pol/.cache/nixpkgs-review/pr-337062/attrs.nix

Link to currently reviewing PR:
https://github.com/NixOS/nixpkgs/pull/337062

38 packages built:
linuxKernel.packages.linux_4_19.nvidia_x11_legacy390 linuxKernel.packages.linux_4_19.nvidia_x11_legacy390.bin linuxKernel.packages.linux_4_19_hardened.nvidia_x11_legacy390 linuxKernel.packages.linux_4_19_hardened.nvidia_x11_legacy390.bin linuxKernel.packages.linux_5_10.nvidia_x11_legacy390 linuxKernel.packages.linux_5_10.nvidia_x11_legacy390.bin linuxKernel.packages.linux_5_10_hardened.nvidia_x11_legacy390 linuxKernel.packages.linux_5_10_hardened.nvidia_x11_legacy390.bin linuxKernel.packages.linux_5_15.nvidia_x11_legacy390 linuxKernel.packages.linux_5_15.nvidia_x11_legacy390.bin linuxKernel.packages.linux_5_15_hardened.nvidia_x11_legacy390 linuxKernel.packages.linux_5_15_hardened.nvidia_x11_legacy390.bin linuxKernel.packages.linux_5_4.nvidia_x11_legacy390 linuxKernel.packages.linux_5_4.nvidia_x11_legacy390.bin linuxKernel.packages.linux_5_4_hardened.nvidia_x11_legacy390 linuxKernel.packages.linux_5_4_hardened.nvidia_x11_legacy390.bin linuxKernel.packages.linux_6_1.nvidia_x11_legacy390 linuxKernel.packages.linux_6_1.nvidia_x11_legacy390.bin linuxKernel.packages.linux_6_10.nvidia_x11_legacy390 linuxKernel.packages.linux_6_10.nvidia_x11_legacy390.bin linuxKernel.packages.linux_6_1_hardened.nvidia_x11_legacy390 linuxKernel.packages.linux_6_1_hardened.nvidia_x11_legacy390.bin linuxKernel.packages.linux_6_6.nvidia_x11_legacy390 linuxKernel.packages.linux_6_6.nvidia_x11_legacy390.bin linuxKernel.packages.linux_hardened.nvidia_x11_legacy390 linuxKernel.packages.linux_hardened.nvidia_x11_legacy390.bin linuxKernel.packages.linux_latest_libre.nvidia_x11_legacy390 linuxKernel.packages.linux_latest_libre.nvidia_x11_legacy390.bin linuxKernel.packages.linux_libre.nvidia_x11_legacy390 linuxKernel.packages.linux_libre.nvidia_x11_legacy390.bin linuxKernel.packages.linux_lqx.nvidia_x11_legacy390 linuxKernel.packages.linux_lqx.nvidia_x11_legacy390.bin linuxKernel.packages.linux_xanmod.nvidia_x11_legacy390 linuxKernel.packages.linux_xanmod.nvidia_x11_legacy390.bin linuxKernel.packages.linux_xanmod_latest.nvidia_x11_legacy390 linuxKernel.packages.linux_xanmod_latest.nvidia_x11_legacy390.bin linuxKernel.packages.linux_zen.nvidia_x11_legacy390 linuxKernel.packages.linux_zen.nvidia_x11_legacy390.bin

$ /nix/store/hvkjjva2knprwmqaa989wrxhlw3zc19v-nix-2.18.5/bin/nix-shell --argstr system x86_64-linux --argstr nixpkgs-path /home/pol/.cache/nixpkgs-review/pr-337062/nixpkgs --argstr nixpkgs-config-path /tmp/tmp129cio38.nix --argstr attrs-path /home/pol/.cache/nixpkgs-review/pr-337062/attrs.nix --nix-path 'nixpkgs=/home/pol/.cache/nixpkgs-review/pr-337062/nixpkgs nixpkgs-overlays=/tmp/tmpryw4j5el' /nix/store/s5dnqifxhlqrd8zicabxswwdgjlhplvq-nixpkgs-review-2.10.5/lib/python3.12/site-packages/nixpkgs_review/nix/review-shell.nix

[nix-shell:~/.cache/nixpkgs-review/pr-337062]$
```

**Comment by Kiskae** [comment]
> Result of `nixpkgs-review pr 337062` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)

`--extra-nixpkgs-config '{ nvidia.acceptLicense = true; }' --eval local` is required to make it evaluate older drivers due to the `acceptLicense` check. Ironically it doesn't work for newer drivers.

**Comment by drupol** [comment]
Result of `nixpkgs-review pr 337062 --extra-nixpkgs-config '{ nvidia.acceptLicense = true; }' --checkout commit` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>38 packages built:</summary>
  <ul>
    <li>linuxKernel.packages.linux_4_19.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_4_19.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_4_19_hardened.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_4_19_hardened.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_5_10.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_5_10.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_5_10_hardened.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_5_10_hardened.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_5_15.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_5_15.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_5_15_hardened.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_5_15_hardened.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_5_4.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_5_4.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_5_4_hardened.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_5_4_hardened.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_6_1.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_6_1.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_6_10.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_6_10.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_6_1_hardened.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_6_1_hardened.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_6_6.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_6_6.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_hardened.nvidia_x11_legacy390 (linuxKernel.packages.linux_6_6_hardened.nvidia_x11_legacy390)</li>
    <li>linuxKernel.packages.linux_hardened.nvidia_x11_legacy390.bin (linuxKernel.packages.linux_6_6_hardened.nvidia_x11_legacy390.bin)</li>
    <li>linuxKernel.packages.linux_latest_libre.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_latest_libre.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_libre.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_libre.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_lqx.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_lqx.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_xanmod.nvidia_x11_legacy390.bin</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_legacy390 (linuxKernel.packages.linux_xanmod_stable.nvidia_x11_legacy390)</li>
    <li>linuxKernel.packages.linux_xanmod_latest.nvidia_x11_legacy390.bin (linuxKernel.packages.linux_xanmod_stable.nvidia_x11_legacy390.bin)</li>
    <li>linuxKernel.packages.linux_zen.nvidia_x11_legacy390</li>
    <li>linuxKernel.packages.linux_zen.nvidia_x11_legacy390.bin</li>
  </ul>
</details>

File: pkgs/os-specific/linux/nvidia-x11/default.nix [hunk 1]
```diff
@@ -128,14 +128,33 @@ rec {
   };
 
   # Last one supporting x86
-  legacy_390 = generic {
+  legacy_390 = let
+  # Source corresponding to https://aur.archlinux.org/packages/nvidia-390xx-dkms
```
Comment (Kiskae) [inline pkgs/os-specific/linux/nvidia-x11/default.nix:132]:
```suggestion
    # Source corresponding to https://aur.archlinux.org/packages/nvidia-390xx-dkms
```

Lets not get bonked by the code formatter

File: pkgs/os-specific/linux/nvidia-x11/default.nix [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{ lib, callPackage, fetchFromGitHub, fetchurl, fetchpatch, stdenv, pkgsi686Linux }:
+{ lib, callPackage, fetchFromGitHub, fetchgit, fetchpatch, stdenv, pkgsi686Linux }:
```
Comment (Kiskae) [inline pkgs/os-specific/linux/nvidia-x11/default.nix:1]: Had to check if we didn't use fetchurl, nice catch.

**Review by Kiskae — APPROVED** [review]

**Pull request merged**